downloader/src/main/kotlin/Downloader.kt

```diff
@@ -335,6 +340,12 @@ class Downloader {
                         log.warn { "Unable to delete temporary directory '$gemDirectory'." }
                     }
                 }
+            } else if (target.id.type == "Cargo") {
+                // without following redirect Cargo's package do not have an extension
```

Comment:
Which redirect? HTTP? Should we instead configure our HTTP client to follow redirects?

Reply:
I solved this by using for the name of the archive the last bit of the url which we get from the last redirect. This block is gone now.